File: AnyCode/InsertSort.c
#include <stdio.h>

void InsertSort(int *m1, int size)
{
    int j, i, key;
    
    for(j = 1; j <= (size-1); j++){
        key = m1[j];
        
        i = j - 1;
        while(i > -1 && m1[i] > key){
            m1[i+1] = m1[i];
            i -= 1;
        }
        m1[i+1] = key;
    }
}

int main()
{
    int a[] = {5, 2, 4, 6, 1, 3};
    InsertSort(a, 6);
    
    for(int i = 0; i <= 5; i++){
        printf("%d ", a[i]);
    }
    
    return 0;
}

File: AnyCode/InsertSort.py
def InsertSort(A):
    
    for j in range(1, len(A)):
        key = A[j];
        
        i = j - 1;
        while(i > -1 and A[i] > key):
            A[i+1] = A[i];
            i -= 1;

        A[i+1] = key;

    return A;
    
    
sortm = InsertSort([5, 2, 4, 6, 1, 3]);
print(sortm);